app/components/post_list/index.js

```diff
@@ -5,13 +5,20 @@ import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
 import {refreshChannelWithRetry} from 'app/actions/views/channel';
+import {makePreparePostIdsForPostList} from 'app/selectors/post_list';
+
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
 import PostList from './post_list';
 
-function mapStateToProps(state) {
-    return {
-        theme: getTheme(state)
+function makeMapStateToProps() {
+    const preparePostIds = makePreparePostIdsForPostList();
+
+    return (state, ownProps) => {
+        return {
+            postIds: preparePostIds(state, ownProps),
```

Comment:
this selector is getting executed every time the state changes and it looks like an expensive one.

If you get a post in another channel that you're not seeing this selector will execute again, yes the result is the memoized result but there are JS cycles spent here

Reply:
perhaps this should be called if the postIds for the current view have changed?

Reply:
I'll look to see what I can do, but I'm not convinced that it's that expensive. Yes, it's taking more cycles, but it's still better than the old version, and preventing re-renders is likely far more impactful